hab_gui/cli.py

```diff
@@ -3,31 +3,42 @@
 import click
 from hab.cli import UriArgument, UriHelpClass
 
-import hab_gui.utils
+import hab_gui.utils as utils
+
+from .widgets.splash_screen import SplashScreen
 
 logger = logging.getLogger(__name__)
 
 
-def get_application(settings=None, **kwargs):
+def get_application(settings=None, splash=True, **kwargs):
     """Returns the QApplication instance, creating it if required.
 
     If settings is passed, then the `hab_gui_init` entry point is processed, any
     other kwargs are passed to `cli_args` of `hab_gui.utils.entry_point_init`.
     """
+    # TODO update doc string
     from Qt.QtWidgets import QApplication
 
     global app
 
     if settings:
-        hab_gui.utils.entry_point_init(settings.resolver, "launch", cli_args=kwargs)
+        utils.entry_point_init(settings.resolver, "launch", cli_args=kwargs)
 
     # Get the existing app if possible
     app = QApplication.instance()
+    _splash = None
     if not app:
         # Otherwise create a new QApplication instance
         app = QApplication([])
+        if splash:
+            try:
+                splash_image = utils.get_splash_image(settings.resolver)
+                _splash = SplashScreen(splash_image)
+                _splash.show()
+            except Exception:
+                print("No splash screen configured")
```

Comment:
This is too generic, there should be a fairly specific exception being raised if no splashscreens are defined. Also having a print here essentially makes configuring splashscreens required so you don't always get that message being printed.